### ISSUE TITLE
Support model loading via torch.hub and dotted import paths

### ISSUE THREAD

**Issue by stanlrt** [CAIIVS/raitap#34]
## Summary

RAITAP currently supports model loading from:

- local `.pt` / `.pth` files
- built-in `torchvision.models` names such as `resnet18`

This is useful for demos, but it is still restrictive for users who want to load models from reusable Python packages or well-known model hubs without first exporting full serialized files.

## Problem

Two common workflows are not currently supported:

1. Loading models via `torch.hub`
2. Loading models via a dotted Python import path

Today, `model.source` is effectively interpreted as either:

- a local file path, or
- a `torchvision.models` attribute name

That means users cannot easily express things like:

- "load `resnet18` from `pytorch/vision` through `torch.hub`"
- "import `my_project.models:create_model` and instantiate it"

This also makes state-dict based workflows awkward, because reconstructing a model usually requires an architecture factory or importable builder function.

## Proposed support

Add two new model-source mechanisms.

### 1. `torch.hub` sources

Allow config to specify a Hub repo and entrypoint, for example:

```yaml
model:
  kind: torch_hub
  repo: pytorch/vision
  entrypoint: resnet18
  kwargs:
    weights: DEFAULT
```

Potential implementation details:

- call `torch.hub.load(repo_or_dir=..., model=entrypoint, **kwargs)`
- validate the returned object is an `nn.Module`
- move the model to the resolved device and set `eval()`

### **2. Dotted import-path sources**

Allow config to specify an importable callable or object path, for example:

model:

kind: python

target: my\_package.models:create\_model

kwargs:

num\_classes: 8

Potential implementation details:

- resolve `module:attribute` or dotted-path syntax
- if the target is callable, call it with `kwargs`
- if the result is an `nn.Module`, use it directly
- otherwise raise a clear validation error

## **Why this helps**

- reduces reliance on fragile serialized full-model files
- supports package-based model definitions cleanly
- makes it easier to integrate research code and internal model libraries
- creates a natural path for future state-dict support, where architecture and weights can be specified separately

## **Out of scope for a first pass**

- arbitrary checkpoint restoration from unknown formats
- automatic state-dict reconstruction without an explicit architecture factory
- support for every possible external model registry

## **Acceptance criteria**

- users can load a model from `torch.hub` via config
- users can load a model from an importable Python path via config
- invalid targets produce actionable error messages
- loaded PyTorch models are moved to the resolved device and switched to eval mode
- tests cover happy paths and invalid configuration cases
- docs clearly explain the new source kinds and when to use each one
